src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -221,3 +237,9 @@ class ModelFilter:
     task: Optional[Union[str, List[str]]] = None
     trained_dataset: Optional[Union[str, List[str]]] = None
     tags: Optional[Union[str, List[str]]] = None
+
+    def __post_init__(self):
+        warnings.warn(
+            "ModelFilter is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to the `list_models` method.",
```

Comment:
To make it work with `@expect_deprecation` in the tests

```suggestion
            "'ModelFilter' is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to the `list_models` method.",
```